{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "from pathlib import Path\n",
    "\n",
    "import sys\n",
    "\n",
    "import qiime2\n",
    "import pandas as pd\n",
    "import statsmodels.api as sm\n",
    "import numpy as np\n",
    "import statsmodels.formula.api as smf\n",
    "import seaborn as sns\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "# sys.path.append('/Users/alchemist/Aakash/Diet/gut_brain_project/pipeline_trail_version/dokdo/')\n",
    "# import api\n",
    "import statsmodels.api as sm\n",
    "import scipy\n",
    "\n",
    "from tempfile import mkdtemp\n",
    "from qiime2 import Artifact\n",
    "from qiime2.plugins import demux, deblur, quality_filter, \\\n",
    "                          metadata, feature_table, alignment, \\\n",
    "                          phylogeny, diversity, emperor, feature_classifier, \\\n",
    "                          taxa, composition, dada2, vsearch, longitudinal, \\\n",
    "                          gneiss, songbird, aldex2, qurro, diversity_lib\n",
    "from qiime2.plugins.diversity.pipelines import alpha,alpha_phylogenetic,beta,beta_phylogenetic\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "\n",
    "pd.options.display.max_seq_items = 2000"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before beginning this tutorial, create a new directory and change to that directory.  \n",
    "**make sure to change the path in `workdir`**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Input the variables below: "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-05-18T08:12:24.314480Z",
     "start_time": "2021-05-18T08:12:24.311462Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/alchemist/Aakash/BonnGut/Microbiome_pipeline'"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "workdir = !pwd\n",
    "workdir = str(Path(workdir[0]).parents[0])\n",
    "workdir"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Sample metadata"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### read  metadata into `qiime2.Metadata` object"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-05-18T08:12:24.398332Z",
     "start_time": "2021-05-18T08:12:24.326382Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>barcode-sequence</th>\n",
       "      <th>body-site</th>\n",
       "      <th>year</th>\n",
       "      <th>month</th>\n",
       "      <th>day</th>\n",
       "      <th>Person_ID</th>\n",
       "      <th>reported-antibiotic-usage</th>\n",
       "      <th>days-since-experiment-start</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample-id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>L1S8</th>\n",
       "      <td>AGCTGACTAGTC</td>\n",
       "      <td>gut</td>\n",
       "      <td>2008.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>subject-1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L1S57</th>\n",
       "      <td>ACACACTATGGC</td>\n",
       "      <td>gut</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>subject-1</td>\n",
       "      <td>No</td>\n",
       "      <td>84.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L1S76</th>\n",
       "      <td>ACTACGTGTGGT</td>\n",
       "      <td>gut</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>subject-1</td>\n",
       "      <td>No</td>\n",
       "      <td>112.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L1S105</th>\n",
       "      <td>AGTGCGATGCGT</td>\n",
       "      <td>gut</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>subject-1</td>\n",
       "      <td>No</td>\n",
       "      <td>140.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L2S155</th>\n",
       "      <td>ACGATGCGACCA</td>\n",
       "      <td>left palm</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>subject-1</td>\n",
       "      <td>No</td>\n",
       "      <td>84.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          barcode-sequence  body-site    year  month   day  Person_ID  \\\n",
       "sample-id                                                               \n",
       "L1S8          AGCTGACTAGTC        gut  2008.0   10.0  28.0  subject-1   \n",
       "L1S57         ACACACTATGGC        gut  2009.0    1.0  20.0  subject-1   \n",
       "L1S76         ACTACGTGTGGT        gut  2009.0    2.0  17.0  subject-1   \n",
       "L1S105        AGTGCGATGCGT        gut  2009.0    3.0  17.0  subject-1   \n",
       "L2S155        ACGATGCGACCA  left palm  2009.0    1.0  20.0  subject-1   \n",
       "\n",
       "          reported-antibiotic-usage  days-since-experiment-start  \n",
       "sample-id                                                         \n",
       "L1S8                            Yes                          0.0  \n",
       "L1S57                            No                         84.0  \n",
       "L1S76                            No                        112.0  \n",
       "L1S105                           No                        140.0  \n",
       "L2S155                           No                         84.0  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_metadata    = qiime2.Metadata.load(workdir+'/metadata/metadata.tsv')\n",
    "sample_metadata_pd = sample_metadata.to_dataframe()\n",
    "sample_metadata_pd.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Loading Feature table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-05-18T08:12:27.873089Z",
     "start_time": "2021-05-18T08:12:24.441088Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/var/folders/bj/4d5l6xgx2fjgpp5jy9ql6jfh0000gn/T/qiime2-archive-h63butmh')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 3b1c3175-256b-44d3-8c46-35fedcb0e4fa>"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYUAAAEKCAYAAAD9xUlFAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAVRUlEQVR4nO3de9AldX3n8fcHBEWEcRVMKRcHHYKyq/EySOJaRrJKQDOgLKKgSTAsIykxcbNawIpi1kqxVC6VkFV0NGZMNkBgNMhEDF4iEl3EGW7KRUrCgoy4AiYZLkYJ8N0/Tk9znHou/Vz6nOc8835VnXrO+Z3T3d/nx3A+T/ev+9epKiRJAthp3AVIkpYOQ0GS1DIUJEktQ0GS1DIUJEktQ0GS1HrCuAtYiL322qtWrlw57jIkaaJcc80191XV3lO9N5GhkGQNsGbVqlVs3rx53OVI0kRJcud0703k4aOq2lhVa1esWDHuUiRpWZnIUJAk9WMiQyHJmiTrtm7dOu5SJGlZmchQ8PCRJPVjIkNBktQPQ0GS1JrIUHBMQZL6MZHXKVTVRmDj6tWrT57vOs6/+ruLWJG0eE44dP9xl6Ad2ETuKUiS+mEoSJJahoIkqTWRoeBAsyT1YyJDwYvXJKkfExkKkqR+GAqSpJahIElqGQqSpNZEhoJnH0lSPyYyFDz7SJL6MZGhIEnqh6EgSWoZCpKklqEgSWpNZCh49pEk9WMiQ8GzjySpHxMZCpKkfhgKkqSWoSBJahkKkqSWoSBJahkKkqSWoSBJak1kKHjxmiT1YyJDwYvXJKkfExkKkqR+GAqSpJahIElqGQqSpJahIElqGQqSpJahIElqGQqSpJahIElqGQqSpNZEhoJzH0lSPyYyFJz7SJL6MZGhIEnqh6EgSWoZCpKklqEgSWoZCpKklqEgSWoZCpKklqEgSWoZCpKklqEgSWoZCpKklqEgSWoZCpKklqEgSWrNGgpJ3phkj+b5mUk+neQl/ZcmSRq1LnsK76uqB5K8Avhl4JPAef2WJUkahy6h8Gjz83XAeVX1GWDXPopJ8vokH0vymSSH97ENSdL0uoTC95J8FDgOuCzJEzsuB0CSTyS5J8mN27UfkeTWJLclOR2gqi6pqpOBE4E3df4tJEmLosuX+3HA5cARVfUvwNOA98xhG+uBI4YbkuwMfAg4EjgYOD7JwUMfObN5X5I0QrOGQlX9CLgHeEXT9Ajwna4bqKorgX/arvllwG1VdXtVPQxcCBydgXOAz1XVtVOtL8naJJuTbL733nu7liFJ6qDL2UdnAacBZzRNuwD/e4Hb3Qe4a+j1lqbtncCrgWOTnDLVglW1rqpWV9Xqvffee4FlSJKGPaHDZ94AvBi4FqCq7t52iuoCZIq2qqpzgXMXuG5J0jx1GVN4uKoKKIAkuy/CdrcA+w293he4exHWK0lagC6hcFFz9tFTk5wMfBH42AK3uwk4MMkBSXYF3gxc2nXhJGuSrNu6desCy5AkDZsxFJIE+GtgA/Ap4CDg/VX1p103kOQC4CrgoCRbkpxUVY8ApzI4q+kW4KKquqnrOqtqY1WtXbFiRddFJEkdzDimUFWV5JKqeinwhflsoKqOn6b9MuCy+axTktSPLoePvp7kkN4rmQMPH0lSP7qEwmHAVUn+Mck3k3wryTf7LmwmHj6SpH50OSX1yN6rkCQtCV1CoXqvQpK0JHQJhc8yCIYATwIOAG4F/n2Pdc0oyRpgzapVq8ZVgiQtS13mPnpBVb2w+Xkgg3mLvtp/aTPW5JiCJPVgzrfjbCaqW1JnI0mSFsesh4+S/M7Qy52AlwBOTypJy1CXMYXhye8eYTDG8Kl+ypEkjVOXULi5qi4ebkjyRuDiaT7fOweaJakfXcYUzujYNjIONEtSP6bdU0hyJPBaYJ8kw/c42JPBYSRJ0jIz0+Gju4HNwFHANUPtDwD/tc+iJEnjMW0oVNUNwA1Jzq+qfxthTZKkMekyprAyyYYkNye5fduj98pm4CypktSPLqHw58B5DMYRDgP+AvjLPouajQPNktSPLqGwW1V9CUhV3VlVHwB+qd+yJEnj0OU6hR8n2Qn4TpJTge8Bz+i3LEnSOHTZU3gX8GTgt4CXAm8Ffr3PoiRJ4zHrnkJVbQJIUlX1tv5LkiSNy6x7Ckl+IcnNwC3N659L8uHeK5MkjVyXw0d/DPwy8ENor194ZZ9FzcZTUiWpH53up1BVd23X9GgPtXTmKamS1I8uZx/dleTlQCXZlcGA8y39liVJGocuewqnAO8A9gG2AC9qXkuSlpmZZkk9p6pOAw6rqreMsCZJ0pjMtKfw2iS7MOZ7J0iSRmemMYW/A+4Ddk9yPxCgtv2sqj1HUJ8kaYSm3VOoqvdU1Qrgs1W1Z1XtMfxzhDVKkkZk1oHmqjp6FIVIksavyympS06SNcCaVatWjbsUadGdf/V3e13/CYfu3+v6Ndk6Xby21HjxmiT1Y9pQSPKl5uc5oytHkjROMx0+emaSXwSOSnIhg7OOWlV1ba+VSZJGbqZQeD9wOrAv8EfbvVd49zVJWnamDYWq2gBsSPK+qvrgCGuSJI1Jl5vsfDDJUTw+XfYVVfW3/ZYlSRqHLjfZORv4beDm5vHbTZskaZnpcp3C64AXVdVjAEk+CVyHcyJJ0rLT9TqFpw499+IASVqmuuwpnA1cl+TLDE5LfSXuJUjSstRloPmCJFcAhzAIhdOq6v/1XZikfixkGg2nyFj+Os19VFXfBy7tuZbOnPtIkvrh3EeSpNZEhoIkqR8zhkKSnZLcOKpiJEnjNWMoNNcm3JDE0SVJ2gF0GWh+JnBTkm8AD21rrKqjeqtKkjQWXULhd3uvQpK0JHS5TuErSZ4NHFhVX0zyZGDn/kuTJI1alwnxTgY2AB9tmvYBLumzKEnSeHQ5JfUdwH8E7geoqu8Az+izKEnSeHQJhZ9U1cPbXiR5AoM7r0mSlpkuofCVJP8d2C3Ja4CLgY39liVJGocuoXA6cC/wLeDtwGXAmX0WJUkajy5nHz3W3FjnagaHjW6tKg8fSdIyNGsoJHkd8BHgHxlMnX1AkrdX1ef6Lk6SNFpdLl77Q+CwqroNIMlzgc8ChoIkLTNdQuGebYHQuB24p6d6JC1hXW/Q4814Jte0oZDkmObpTUkuAy5iMKbwRmDTCGqTJI3YTHsKa4ae/wD4xeb5vcC/W+xCkjwHeC+woqqOXez1S5JmN20oVNXbFrryJJ8AfoXBIaj/MNR+BPAnDOZQ+nhV/c+quh04KcmGhW5XkjQ/Xc4+OgB4J7By+PMdp85eD/wv4C+G1rcz8CHgNcAWYFOSS6vq5rkULklafF0Gmi8B/ozBVcyPzWXlVXVlkpXbNb8MuK3ZMyDJhcDRQKdQSLIWWAuw//4OZknSYuoSCj+uqnMXcZv7AHcNvd4CHJrk6cDvAS9OckZVnT3VwlW1DlgHsHr1ai+ik6RF1CUU/iTJWcDngZ9sa6yqa+e5zUzRVlX1Q+CUea5TkrQIuoTCC4BfBX6Jxw8fVfN6PrYA+w293he4ey4rSLIGWLNq1ap5liBJmkqXUHgD8Jzh6bMXaBNwYDOA/T3gzcAJc1lBVW0ENq5evfrkRapJkkS3WVJvAJ46n5UnuQC4CjgoyZYkJ1XVI8CpwOXALcBFVXXTfNYvSVpcXfYUfgb4dpJN/PSYwqynpFbV8dO0X8ZgCm5J0hLSJRTO6r2KOXJMQVraZpsjybmRlq4u91P4yigKmQvHFCSpH12uaH6Ax+/JvCuwC/BQVe3ZZ2GSpNHrsqewx/DrJK9ncFWyJGmZ6XL20U+pqkuY/zUKiyLJmiTrtm7dOs4yJGnZ6XL46JihlzsBq3n8cNJYOKYgSf3ocvbR8H0VHgHuYDCBnSRpmekyprDg+ypIkibDTLfjfP8My1VVfbCHeiRJYzTTQPNDUzwATgJO67muGTnQLEn9mDYUquoPtz0Y3L9gN+BtwIXAc0ZU33S1bayqtStWrBhnGZK07Mw4ppDkacDvAG8BPgm8pKr+eRSFSZJGb6Yxhd8HjmGwl/CCqnpwZFVJksZipjGF/wY8CzgTuDvJ/c3jgST3j6Y8SdIoTbunUFVzvtp5VJwlVZL6sWS/+GfiQLMk9WMiQ0GS1A9DQZLUMhQkSS1DQZLUMhQkSa2JDAXnPpKkfkxkKHhKqiT1YyJDQZLUD0NBktQyFCRJLUNBktQyFCRJLUNBktQyFCRJrRlvx7lUeT8FabKdf/V3p2w/4dD9R1yJtjeRewpevCZJ/ZjIUJAk9cNQkCS1DAVJUstQkCS1DAVJUstQkCS1DAVJUstQkCS1DAVJUstQkCS1nPtI0pIx3ZxIs3HOpMUzkXsKzn0kSf2YyFCQJPXDUJAktQwFSVLLUJAktQwFSVLLUJAktQwFSVLLUJAktQwFSVLLUJAktQwFSVLLUJAktQwFSVLLUJAktQwFSVLLUJAktZbMndeS7A58GHgYuKKq/mrMJUnSDqfXPYUkn0hyT5Ibt2s/IsmtSW5LcnrTfAywoapOBo7qsy5J0tT6Pny0HjhiuCHJzsCHgCOBg4HjkxwM7Avc1Xzs0Z7rkiRNodfDR1V1ZZKV2zW/DLitqm4HSHIhcDSwhUEwXM8MYZVkLbAWYP/9vVm3JDj/6u9O+94Jh+4/6+eGPzPduqdaz2xtXerdfh0nHLp/+3Om5btsZz7GMdC8D4/vEcAgDPYBPg385yTnARunW7iq1lXV6qpavffee/dbqSTtYMYx0Jwp2qqqHgLeNupiJEmPG8eewhZgv6HX+wJ3z2UFSdYkWbd169ZFLUySdnTjCIVNwIFJDkiyK/Bm4NK5rKCqNlbV2hUrVvRSoCTtqPo+JfUC4CrgoCRbkpxUVY8ApwKXA7cAF1XVTX3WIUnqpu+zj46fpv0y4LI+ty1JmruJnObCMQVJ6sdEhoJjCpLUj4kMBUlSP1JV465h3pLcC9w5z8VXAPM9/tR12dk+N937U7V3adv+9V7AfR3qnC/7cOHsw4WbtD6cqn3Uffjsqpr66t+q2iEfwLq+l53tc9O9P1V7l7YpXm+2D+1D+3Bp9WHHPhtpHw4/duTDR9NOpbGIy872uenen6q9S9tCfqf5sA8Xzj5cuEnrw6nax92HrYk+fKSZJdlcVavHXccksw8Xzj5cuFH24Y68p7AjWDfuApYB+3Dh7MOFG1kfuqcgSWq5pyBJahkKkqSWoSBJahkKO5Akz0nyZ0k2jLuWSZXk9Uk+luQzSQ4fdz2TKMnzk3wkyYYkvznueiZVkt2TXJPkVxZzvYbChEvyiST3JLlxu/Yjktya5LYkpwNU1e1VddJ4Kl265tiHl1TVycCJwJvGUO6SNMc+vKWqTgGOAzxVtTGXPmycBly02HUYCpNvPXDEcEOSnYEPAUcCBwPHJzl49KVNjPXMvQ/PbN7XwHrm0IdJjgK+CnxptGUuaevp2IdJXg3cDPxgsYswFCZcVV0J/NN2zS8Dbmv2DB4GLgSOHnlxE2IufZiBc4DPVdW1o651qZrrv8OqurSqXg68ZbSVLl1z7MPDgJ8HTgBOTrJo3+W93mRHY7MPcNfQ6y3AoUmeDvwe8OIkZ1TV2WOpbjJM2YfAO4FXAyuSrKqqj4yjuAkx3b/DVwHHAE/Em23NZso+rKpTAZKcCNxXVY8t1gYNheUpU7RVVf0QOGXUxUyo6frwXODcURczoabrwyuAK0ZbysSasg/bJ1XrF3uDHj5anrYA+w293he4e0y1TCr7cOHsw4UbeR8aCsvTJuDAJAck2RV4M3DpmGuaNPbhwtmHCzfyPjQUJlySC4CrgIOSbElyUlU9ApwKXA7cAlxUVTeNs86lzD5cOPtw4ZZKHzohniSp5Z6CJKllKEiSWoaCJKllKEiSWoaCJKllKEiSWoaCxirJo0muH3qsHHdNS1GS5zX9c12S585j+XcleXIftWl58ToFjVWSB6vqKdO8Fwb/Rhdtsq9JkWTnqnp06PXpwG5VddY813cHsLqq7pvDMk9oLp7SDsQ9BS0pSVYmuSXJh4Frgf2SvCfJpiTfTPK7Q599b3PzkS8muSDJu5v2K5Ksbp7v1XwhkmTnJL8/tK63N+2vapbZkOTbSf6qCSSSHJLk/yS5Ick3kuyR5B+SvGiojq8leeF2v8eJGdyd7e+aGs8aeu+tzbquT/LRZs58kjyY5H8kuRr4haHPvxZ4F/Bfknx5lnWcl2Rzkpu29VWS3wKeBXx5aPkHh9Z/bJL1zfP1Sf6o+dw5SZ7b/A7XNL/38+b/X1cToap8+BjbA3gUuL55/A2wEngM+Pnm/cOBdQxmi9wJ+FvglcBLgW8BTwb2BG4D3t0scwWDv4oB9gLuaJ6vBc5snj8R2AwcALwK2MpgsrGdGEw18ApgV+B24JBmmT0ZzCz868AfN20/C2ye4vc6Efg+8HRgN+BGBncZez6wEdil+dyHgV9rnhdw3DT99IGh32+mdTyt+blz0w8vbF7fAew1tL4Hh54fC6xvnq9v+njn5vWXgAOb54cCfz/ufzM++n04dbbG7V+raviv7pXAnVX19abp8OZxXfP6KcCBwB7A31TVj5rlukwSdjjwwiTHNq9XNOt6GPhGVW1p1nU9g3DaCny/qjYBVNX9zfsXA+9L8h7gNxh8kU7lCzWYrpwkn2YQNI8wCLRNzc7IbsA9zecfBT7V4ff4TzOs47gkaxmE1zMZ3K3rmx3WOeziqno0yVOAlwMXN9uBQZhqGTMUtBQ9NPQ8wNlV9dHhDyR5F0Pzym/nER4/NPqk7db1zqq6fLt1vQr4yVDTowz+38hU26iqHyX5AoM7YM10n+Htl61mnZ+sqjOm+PyPa2gcYQZTriPJAcC7GezZ/HNzSOhJUyy/fW3bf2Zb/+8E/MtwaGv5c0xBS93lwG80f7WSZJ8kzwCuBN6QZLckewBrhpa5g8Ff0jA4NDK8rt9Mskuzrp9NsvsM2/428KwkhzSf3yPJtj+kPs7gZjubqmr7Wyhu85okT0uyG/B64GsMDscc2/wONO8/e/Zu+CnTrWNPBl/oW5P8DIP7+m7zAIO9q21+kOT5GdzG8Q1TbaTZM/q/Sd7YbCdJfm6OtWrCuKegJa2qPp/k+cBVzSGMB4G3VtW1Sf6awVjEncA/DC32B8BFSX4V+Puh9o8zOCx0bTOQfC+DL+vptv1wkjcBf9p8sf8rg1txPlhV1yS5H/jzGcr/KvCXwCrg/KraDJDkTODzzRfyvwHvaH6HTqrq5qnWUVVfT3IdcBODsZCvDS22Dvhcku9X1WHA6QzGDu5iMN4x5RlgDO6hfF6zvV0Y3CP4hq61avJ4SqqWhSQfYPBl/Qcj2t6zGAzkPq+mOGU2g3vnrq7mXrrSpPDwkTRHSX4NuBp471SBIE0y9xQkSS33FCRJLUNBktQyFCRJLUNBktQyFCRJLUNBktT6/6N3a+049wa8AAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "#\n",
    "table                       = Artifact.load(workdir+'/feature_table/table.qza')\n",
    "rep_seq                     = Artifact.load(workdir+'/rep_seq/rep_seq.qza')\n",
    "phylogenic_tree_rooted_tree = Artifact.load(workdir+'/phylo_tree/rooted_tree.qza')\n",
    "#\n",
    "output_viz = feature_table.visualizers.summarize(table,sample_metadata) \n",
    "output_viz.visualization"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Input the variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "Max_depth         = 9820\n",
    "Min_depth         = 897   #approximation\n",
    "Max_feature_depth = 3909        #depth with maximum features\n",
    "Step              = 2500\n",
    "rarefy_list       = np.arange(Min_depth,Max_depth,Step)\n",
    "rarefy_list       = np.append(rarefy_list,Max_feature_depth)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculating diversity metrics for 897\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/alchemist/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/sklearn/metrics/pairwise.py:1761: DataConversionWarning: Data was converted to boolean for metric jaccard\n",
      "  warnings.warn(msg, DataConversionWarning)\n",
      "/Users/alchemist/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:152: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.013522465080301867 and the largest is 3.787971034135235.\n",
      "  RuntimeWarning\n",
      "/Users/alchemist/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:152: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.010891490933327783 and the largest is 2.6502541279538696.\n",
      "  RuntimeWarning\n",
      "/Users/alchemist/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:152: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.022563051359418735 and the largest is 3.482657885882314.\n",
      "  RuntimeWarning\n",
      "/Users/alchemist/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/sklearn/metrics/pairwise.py:1761: DataConversionWarning: Data was converted to boolean for metric jaccard\n",
      "  warnings.warn(msg, DataConversionWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculating diversity metrics for 3397\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/alchemist/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/sklearn/metrics/pairwise.py:1761: DataConversionWarning: Data was converted to boolean for metric jaccard\n",
      "  warnings.warn(msg, DataConversionWarning)\n",
      "/Users/alchemist/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:152: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.00450240378183084 and the largest is 2.619973921356427.\n",
      "  RuntimeWarning\n",
      "/Users/alchemist/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:152: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0018680699449827524 and the largest is 2.169740886645954.\n",
      "  RuntimeWarning\n",
      "/Users/alchemist/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/sklearn/metrics/pairwise.py:1761: DataConversionWarning: Data was converted to boolean for metric jaccard\n",
      "  warnings.warn(msg, DataConversionWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculating diversity metrics for 5897\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/alchemist/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/sklearn/metrics/pairwise.py:1761: DataConversionWarning: Data was converted to boolean for metric jaccard\n",
      "  warnings.warn(msg, DataConversionWarning)\n",
      "/Users/alchemist/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:152: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0030909390868172314 and the largest is 2.120710335461436.\n",
      "  RuntimeWarning\n",
      "/Users/alchemist/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:152: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0013269662777759196 and the largest is 1.5804289622824366.\n",
      "  RuntimeWarning\n",
      "/Users/alchemist/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/sklearn/metrics/pairwise.py:1761: DataConversionWarning: Data was converted to boolean for metric jaccard\n",
      "  warnings.warn(msg, DataConversionWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculating diversity metrics for 8397\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/alchemist/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/sklearn/metrics/pairwise.py:1761: DataConversionWarning: Data was converted to boolean for metric jaccard\n",
      "  warnings.warn(msg, DataConversionWarning)\n",
      "/Users/alchemist/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/sklearn/metrics/pairwise.py:1761: DataConversionWarning: Data was converted to boolean for metric jaccard\n",
      "  warnings.warn(msg, DataConversionWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculating diversity metrics for 3909\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/alchemist/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/sklearn/metrics/pairwise.py:1761: DataConversionWarning: Data was converted to boolean for metric jaccard\n",
      "  warnings.warn(msg, DataConversionWarning)\n",
      "/Users/alchemist/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:152: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.001319425759144815 and the largest is 1.7986538748823198.\n",
      "  RuntimeWarning\n",
      "/Users/alchemist/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:152: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0022574811956153702 and the largest is 2.1806582624451147.\n",
      "  RuntimeWarning\n",
      "/Users/alchemist/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/sklearn/metrics/pairwise.py:1761: DataConversionWarning: Data was converted to boolean for metric jaccard\n",
      "  warnings.warn(msg, DataConversionWarning)\n"
     ]
    }
   ],
   "source": [
    "for sampling_depth in rarefy_list:\n",
    "    #\n",
    "    print('Calculating diversity metrics for '+str(sampling_depth))\n",
    "    #\n",
    "    Path(str(workdir)+'/diversity_analysis/sampling_depth_'+str(sampling_depth)).mkdir(parents=True, exist_ok=True)\n",
    "    #\n",
    "    core_metrics = diversity.pipelines.core_metrics_phylogenetic(table = table,\n",
    "                                                             phylogeny = phylogenic_tree_rooted_tree,\n",
    "                                                             sampling_depth = sampling_depth,\n",
    "                                                             metadata = sample_metadata,\n",
    "                                                             ) \n",
    "    #\n",
    "    table_rarefied = core_metrics.rarefied_table\n",
    "    non_phylo_metric_list = ['observed_features','shannon']\n",
    "    col_names =  ['observed_features', 'shannon', 'Faith']\n",
    "    temp_alpha  = pd.DataFrame(columns = col_names)\n",
    "    for met in non_phylo_metric_list:\n",
    "        alpha_metric = alpha(\n",
    "                       table      =       table_rarefied,\n",
    "                       metric     =       met\n",
    "                             )\n",
    "        Path(str(workdir)+'/diversity_analysis/sampling_depth_'+str(sampling_depth)+'/'+str(met)).mkdir(parents=True, exist_ok=True)\n",
    "        alpha_metric.alpha_diversity.export_data(str(workdir)+'/diversity_analysis/sampling_depth_'+str(sampling_depth)+'/'+str(met))\n",
    "        alpha_data      = pd.read_csv(str(workdir)+'/diversity_analysis/sampling_depth_'+str(sampling_depth)+'/'+str(met)+'/alpha-diversity.tsv',sep='\\t',index_col=[0])\n",
    "        temp_alpha[met] = alpha_data.iloc[:,0]\n",
    "    core_metrics.faith_pd_vector.export_data(str(workdir)+'/diversity_analysis/sampling_depth_'+str(sampling_depth)+'/faith_pd')\n",
    "    faith_data = pd.read_csv(str(workdir)+'/diversity_analysis/sampling_depth_'+str(sampling_depth)+'/faith_pd/alpha-diversity.tsv',sep='\\t',index_col=[0])\n",
    "    temp_alpha['Faith'] = faith_data.iloc[:,0]\n",
    "    temp_alpha.to_csv(str(workdir)+'/diversity_analysis/sampling_depth_'+str(sampling_depth)+'/alpha_diversity.csv')\n",
    "    #\n",
    "    table_rarefied = core_metrics.rarefied_table\n",
    "    beta_metric_list = ['braycurtis','jaccard']\n",
    "    for met in beta_metric_list:\n",
    "        beta_metric = beta(\n",
    "                       table      =       table_rarefied,\n",
    "                       metric     =       met\n",
    "                             )\n",
    "        Path(str(workdir)+'/diversity_analysis/sampling_depth_'+str(sampling_depth)+'/'+str(met)).mkdir(parents=True, exist_ok=True)\n",
    "        beta_metric.distance_matrix.export_data(str(workdir)+'/diversity_analysis/sampling_depth_'+str(sampling_depth)+'/'+str(met))\n",
    "\n",
    "    beta_metric_list = ['unweighted_unifrac','weighted_unifrac']\n",
    "    for met in beta_metric_list:\n",
    "        beta_metric = beta_phylogenetic(\n",
    "                       table      =       table_rarefied,\n",
    "                       metric     =       met,\n",
    "                       phylogeny  = phylogenic_tree_rooted_tree\n",
    "                             )\n",
    "        Path(str(workdir)+'/diversity_analysis/sampling_depth_'+str(sampling_depth)+'/'+str(met)).mkdir(parents=True, exist_ok=True)\n",
    "        beta_metric.distance_matrix.export_data(str(workdir)+'/diversity_analysis/sampling_depth_'+str(sampling_depth)+'/'+str(met))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "List of alpha metrics: 'dominance', 'simpson_e', 'heip_e', 'mcintosh_e',\n",
    "    'simpson', 'singles', 'doubles', 'margalef', 'chao1_ci', 'robbins',\n",
    "    'gini_index', 'shannon', 'brillouin_d', 'fisher_alpha', 'chao1',\n",
    "    'mcintosh_d', 'lladser_pe', 'observed_features', 'enspie',\n",
    "    'kempton_taylor_q', 'ace', 'michaelis_menten_fit', 'esty_ci', 'pielou_e',\n",
    "    'osd', 'berger_parker_d', 'menhinick', 'goods_coverage', 'strong' \\\n",
    "List of beta metric: 'rogerstanimoto', 'kulsinski', 'sokalmichener',\n",
    "    'dice', 'braycurtis', 'matching', 'canberra_adkins', 'jensenshannon',\n",
    "    'sqeuclidean', 'hamming', 'sokalsneath', 'canberra', 'chebyshev',\n",
    "    'seuclidean', 'aitchison', 'yule', 'cosine', 'correlation', 'minkowski',\n",
    "    'euclidean', 'russellrao', 'jaccard', 'cityblock'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>shannon_entropy</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>L1S105</th>\n",
       "      <td>4.004631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L1S140</th>\n",
       "      <td>3.726095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L1S208</th>\n",
       "      <td>4.370518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L1S257</th>\n",
       "      <td>4.740285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L1S281</th>\n",
       "      <td>4.583463</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L1S57</th>\n",
       "      <td>4.126907</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L1S76</th>\n",
       "      <td>3.560113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L1S8</th>\n",
       "      <td>3.079887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L2S155</th>\n",
       "      <td>5.069048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L2S175</th>\n",
       "      <td>4.840492</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L2S204</th>\n",
       "      <td>4.410149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L2S222</th>\n",
       "      <td>6.354954</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L2S240</th>\n",
       "      <td>3.796994</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L2S309</th>\n",
       "      <td>4.568008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L2S357</th>\n",
       "      <td>4.439772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L2S382</th>\n",
       "      <td>5.859370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L3S242</th>\n",
       "      <td>2.875264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L3S294</th>\n",
       "      <td>4.318832</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L3S313</th>\n",
       "      <td>4.446409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L3S341</th>\n",
       "      <td>4.374717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L3S360</th>\n",
       "      <td>5.218470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L3S378</th>\n",
       "      <td>3.606756</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L4S112</th>\n",
       "      <td>5.019651</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L4S137</th>\n",
       "      <td>5.289920</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L4S63</th>\n",
       "      <td>5.992793</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L5S104</th>\n",
       "      <td>3.607041</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L5S155</th>\n",
       "      <td>3.081274</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L5S174</th>\n",
       "      <td>3.278424</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L5S203</th>\n",
       "      <td>3.700270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L5S222</th>\n",
       "      <td>3.374069</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L5S240</th>\n",
       "      <td>3.360759</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L6S20</th>\n",
       "      <td>3.039470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L6S68</th>\n",
       "      <td>3.427174</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L6S93</th>\n",
       "      <td>4.170132</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        shannon_entropy\n",
       "L1S105         4.004631\n",
       "L1S140         3.726095\n",
       "L1S208         4.370518\n",
       "L1S257         4.740285\n",
       "L1S281         4.583463\n",
       "L1S57          4.126907\n",
       "L1S76          3.560113\n",
       "L1S8           3.079887\n",
       "L2S155         5.069048\n",
       "L2S175         4.840492\n",
       "L2S204         4.410149\n",
       "L2S222         6.354954\n",
       "L2S240         3.796994\n",
       "L2S309         4.568008\n",
       "L2S357         4.439772\n",
       "L2S382         5.859370\n",
       "L3S242         2.875264\n",
       "L3S294         4.318832\n",
       "L3S313         4.446409\n",
       "L3S341         4.374717\n",
       "L3S360         5.218470\n",
       "L3S378         3.606756\n",
       "L4S112         5.019651\n",
       "L4S137         5.289920\n",
       "L4S63          5.992793\n",
       "L5S104         3.607041\n",
       "L5S155         3.081274\n",
       "L5S174         3.278424\n",
       "L5S203         3.700270\n",
       "L5S222         3.374069\n",
       "L5S240         3.360759\n",
       "L6S20          3.039470\n",
       "L6S68          3.427174\n",
       "L6S93          4.170132"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alpha_data = pd.read_csv(str(workdir)+'/diversity_analysis/sampling_depth_'+str(sampling_depth)+'/shannon/alpha-diversity.tsv',sep='\\t',index_col=[0])\n",
    "alpha_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "temp_alpha = []\n",
    "alpha_data      = pd.read_csv(str(workdir)+'/diversity_analysis/sampling_depth_'+str(sampling_depth)+'/'+str(met)+'/alpha-diversity.tsv',sep='\\t',index_col=[0])\n",
    "temp_alpha      = temp_alpha.append(alpha_data.iloc[:,0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp_alpha = []\n",
    "temp_alpha"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([3.8323405 , 3.6729354 , 4.47112102, 4.63101647, 4.66813325,\n",
       "       4.1255213 , 3.39258684, 3.25388338, 4.99657172, 4.99982091,\n",
       "       4.35764257, 6.37486462, 3.894814  , 4.57782302, 4.46736162,\n",
       "       5.85031561, 2.87526385, 4.32133295, 4.46220149, 4.32602691,\n",
       "       5.23392584, 3.57582856, 5.26456478, 5.37278694, 6.03270422,\n",
       "       3.62329039, 3.13180923, 3.26744106, 3.72270281, 3.3075086 ,\n",
       "       3.33305559, 3.09603217, 3.36584527, 4.15277199])"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alpha_data.iloc[:,0].values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
