{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "from pathlib import Path\n",
    "\n",
    "import sys\n",
    "\n",
    "import qiime2\n",
    "import pandas as pd\n",
    "import statsmodels.api as sm\n",
    "import numpy as np\n",
    "import statsmodels.formula.api as smf\n",
    "import seaborn as sns\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "# sys.path.append('/Users/alchemist/Aakash/Diet/gut_brain_project/pipeline_trail_version/dokdo/')\n",
    "# import api\n",
    "import statsmodels.api as sm\n",
    "import scipy\n",
    "\n",
    "from tempfile import mkdtemp\n",
    "from qiime2 import Artifact\n",
    "from qiime2.plugins import demux, deblur, quality_filter, \\\n",
    "                          metadata, feature_table, alignment, \\\n",
    "                          phylogeny, diversity, emperor, feature_classifier, \\\n",
    "                          taxa, composition, dada2, vsearch, longitudinal, \\\n",
    "                          gneiss, songbird, aldex2, qurro, diversity_lib\n",
    "from qiime2.plugins.diversity.pipelines import alpha,alpha_phylogenetic,beta,beta_phylogenetic\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "\n",
    "pd.options.display.max_seq_items = 2000"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before beginning this tutorial, create a new directory and change to that directory.  \n",
    "**make sure to change the path in `workdir`**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Input the variables below: "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "workdir = !pwd\n",
    "workdir = str(Path(workdir[0]).parents[0])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# read raw data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "ename": "ValidationError",
     "evalue": "/Users/alchemist/Aakash/BonnGut/Microbiome_pipeline/raw_data is not a file.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValidationError\u001b[0m                           Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-10-ebe8f3e81990>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mraw_sequences\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mqiime2\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mArtifact\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mimport_data\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'SampleData[SequencesWithQuality]'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mraw_data\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mview_type\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'SingleEndFastqManifestPhred33V2'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/qiime2/sdk/result.py\u001b[0m in \u001b[0;36mimport_data\u001b[0;34m(cls, type, view, view_type)\u001b[0m\n\u001b[1;32m    239\u001b[0m         \u001b[0mprovenance_capture\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0marchive\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mImportProvenanceCapture\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mformat_\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmd5sums\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    240\u001b[0m         return cls._from_view(type_, view, view_type, provenance_capture,\n\u001b[0;32m--> 241\u001b[0;31m                               validate_level='max')\n\u001b[0m\u001b[1;32m    242\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    243\u001b[0m     \u001b[0;34m@\u001b[0m\u001b[0mclassmethod\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/qiime2/sdk/result.py\u001b[0m in \u001b[0;36m_from_view\u001b[0;34m(cls, type, view, view_type, provenance_capture, validate_level)\u001b[0m\n\u001b[1;32m    265\u001b[0m         transformation = from_type.make_transformation(to_type,\n\u001b[1;32m    266\u001b[0m                                                        recorder=recorder)\n\u001b[0;32m--> 267\u001b[0;31m         \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtransformation\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mview\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalidate_level\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    268\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    269\u001b[0m         \u001b[0martifact\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcls\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__new__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcls\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/qiime2/core/transform.py\u001b[0m in \u001b[0;36mtransformation\u001b[0;34m(view, validate_level)\u001b[0m\n\u001b[1;32m     66\u001b[0m         \u001b[0;32mdef\u001b[0m \u001b[0mtransformation\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mview\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalidate_level\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'min'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     67\u001b[0m             \u001b[0mview\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcoerce_view\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mview\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 68\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalidate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mview\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalidate_level\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     69\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     70\u001b[0m             \u001b[0mnew_view\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtransformer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mview\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/qiime2/core/transform.py\u001b[0m in \u001b[0;36mvalidate\u001b[0;34m(self, view, level)\u001b[0m\n\u001b[1;32m    141\u001b[0m                             % (view, self._view_type))\n\u001b[1;32m    142\u001b[0m         \u001b[0;31m# Formats have a validate method, so defer to it\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 143\u001b[0;31m         \u001b[0mview\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalidate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlevel\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    144\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    145\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mset_user_owned\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mview\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/qiime2/plugin/model/file_format.py\u001b[0m in \u001b[0;36mvalidate\u001b[0;34m(self, level)\u001b[0m\n\u001b[1;32m     19\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     20\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mis_file\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 21\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mValidationError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"%s is not a file.\"\u001b[0m \u001b[0;34m%\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     22\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     23\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mhasattr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'_validate_'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValidationError\u001b[0m: /Users/alchemist/Aakash/BonnGut/Microbiome_pipeline/raw_data is not a file."
     ]
    }
   ],
   "source": [
    "raw_sequences = qiime2.Artifact.import_data('SampleData[PairedEndSequencesWithQuality]', workdir+'/raw_data',view_type='CasavaOneEightSingleLanePerSampleDirFmt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "raw_sequences.save(workdir+'/demultiplexed_files/demultiplexed.qza') #saving demultipexed data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### read  metadata into `qiime2.Metadata` object"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>barcode-sequence</th>\n",
       "      <th>body-site</th>\n",
       "      <th>year</th>\n",
       "      <th>month</th>\n",
       "      <th>day</th>\n",
       "      <th>Person_ID</th>\n",
       "      <th>reported-antibiotic-usage</th>\n",
       "      <th>days-since-experiment-start</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample-id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>L1S8</th>\n",
       "      <td>AGCTGACTAGTC</td>\n",
       "      <td>gut</td>\n",
       "      <td>2008.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>subject-1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L1S57</th>\n",
       "      <td>ACACACTATGGC</td>\n",
       "      <td>gut</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>subject-1</td>\n",
       "      <td>No</td>\n",
       "      <td>84.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L1S76</th>\n",
       "      <td>ACTACGTGTGGT</td>\n",
       "      <td>gut</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>subject-1</td>\n",
       "      <td>No</td>\n",
       "      <td>112.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L1S105</th>\n",
       "      <td>AGTGCGATGCGT</td>\n",
       "      <td>gut</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>subject-1</td>\n",
       "      <td>No</td>\n",
       "      <td>140.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L2S155</th>\n",
       "      <td>ACGATGCGACCA</td>\n",
       "      <td>left palm</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>subject-1</td>\n",
       "      <td>No</td>\n",
       "      <td>84.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          barcode-sequence  body-site    year  month   day  Person_ID  \\\n",
       "sample-id                                                               \n",
       "L1S8          AGCTGACTAGTC        gut  2008.0   10.0  28.0  subject-1   \n",
       "L1S57         ACACACTATGGC        gut  2009.0    1.0  20.0  subject-1   \n",
       "L1S76         ACTACGTGTGGT        gut  2009.0    2.0  17.0  subject-1   \n",
       "L1S105        AGTGCGATGCGT        gut  2009.0    3.0  17.0  subject-1   \n",
       "L2S155        ACGATGCGACCA  left palm  2009.0    1.0  20.0  subject-1   \n",
       "\n",
       "          reported-antibiotic-usage  days-since-experiment-start  \n",
       "sample-id                                                         \n",
       "L1S8                            Yes                          0.0  \n",
       "L1S57                            No                         84.0  \n",
       "L1S76                            No                        112.0  \n",
       "L1S105                           No                        140.0  \n",
       "L2S155                           No                         84.0  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_metadata    = qiime2.Metadata.load(workdir+'/metadata/metadata.tsv')\n",
    "sample_metadata_pd = sample_metadata.to_dataframe()\n",
    "sample_metadata_pd.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "sample_metadata.save(workdir+'/metadata/metadata.qza')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Demultiplexing sequences"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "demux_sequences = demux.methods.emp_single(single_end_sequences,\n",
    "                                           sample_metadata.get_column('BarcodeSequence'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Looking at quality plots before joining \n",
    "demux_summary = demux.visualizers.summarize(raw_sequences)\n",
    "demux_summary.visualization.export_data(workdir+'/demultiplexed_files/demux_summary')\n",
    "demux_joined.visualization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Looking at quality plots after joining \n",
    "demux_joined_pairs = vsearch.methods.join_pairs(raw_sequences)\n",
    "demux_joined = demux.visualizers.summarize(demux_joined_pairs.joined_sequences)\n",
    "demux_joined.visualization.export_data(workdir+'/demultiplexed_files/demux_joined_summary')\n",
    "demux_joined.visualization"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Denoising using dada2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application(s). This may print messages to stdout and/or stderr.\n",
      "The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: run_dada_paired.R /var/folders/bj/4d5l6xgx2fjgpp5jy9ql6jfh0000gn/T/tmp022078zd/forward /var/folders/bj/4d5l6xgx2fjgpp5jy9ql6jfh0000gn/T/tmp022078zd/reverse /var/folders/bj/4d5l6xgx2fjgpp5jy9ql6jfh0000gn/T/tmp022078zd/output.tsv.biom /var/folders/bj/4d5l6xgx2fjgpp5jy9ql6jfh0000gn/T/tmp022078zd/track.tsv /var/folders/bj/4d5l6xgx2fjgpp5jy9ql6jfh0000gn/T/tmp022078zd/filt_f /var/folders/bj/4d5l6xgx2fjgpp5jy9ql6jfh0000gn/T/tmp022078zd/filt_r 294 240 17 19 2.0 2.0 2 independent consensus 1.0 0 1000000\n",
      "\n"
     ]
    }
   ],
   "source": [
    "sequences = dada2.methods.denoise_paired(raw_sequences,\n",
    "                                  trim_left_f=17, #18\n",
    "                                  trim_left_r=19, #21\n",
    "                                  trunc_len_f=294, #278\n",
    "                                  trunc_len_r=240, #211\n",
    "                                  n_threads=0 \n",
    "                                 )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/alchemist/Aakash/BonnGut/Hannahs_study/qza_files/dada2_stats.qza'"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dada2_stats = sequences.denoising_stats\n",
    "sequences.denoising_stats.save(workdir+\"/dada2_files/dada2_stats.qza\")\n",
    "sequences.denoising_stats.view\n",
    "#help(dada2_stats)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Saving Feature table & rep seq files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/alchemist/Aakash/BonnGut/Hannahs_study/qza_files/rep_seq.qza'"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequences.table.save(workdir+'/feature_table/table.qza')\n",
    "sequences.representative_sequences.save(workdir+'/rep_seq/rep_seq.qza')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Loading Feature table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/var/folders/bj/4d5l6xgx2fjgpp5jy9ql6jfh0000gn/T/qiime2-archive-2vr6ou1y')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 64eeaf22-c8b8-4e92-a5fb-5add82dc3a16>"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYUAAAEKCAYAAAD9xUlFAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAVRUlEQVR4nO3de9AldX3n8fcHBEWEcRVMKRcHHYKyq/EySOJaRrJKQDOgLKKgSTAsIykxcbNawIpi1kqxVC6VkFV0NGZMNkBgNMhEDF4iEl3EGW7KRUrCgoy4AiYZLkYJ8N0/Tk9znHou/Vz6nOc8835VnXrO+Z3T3d/nx3A+T/ev+9epKiRJAthp3AVIkpYOQ0GS1DIUJEktQ0GS1DIUJEktQ0GS1HrCuAtYiL322qtWrlw57jIkaaJcc80191XV3lO9N5GhkGQNsGbVqlVs3rx53OVI0kRJcud0703k4aOq2lhVa1esWDHuUiRpWZnIUJAk9WMiQyHJmiTrtm7dOu5SJGlZmchQ8PCRJPVjIkNBktQPQ0GS1JrIUHBMQZL6MZHXKVTVRmDj6tWrT57vOs6/+ruLWJG0eE44dP9xl6Ad2ETuKUiS+mEoSJJahoIkqTWRoeBAsyT1YyJDwYvXJKkfExkKkqR+GAqSpJahIElqGQqSpNZEhoJnH0lSPyYyFDz7SJL6MZGhIEnqh6EgSWoZCpKklqEgSWpNZCh49pEk9WMiQ8GzjySpHxMZCpKkfhgKkqSWoSBJahkKkqSWoSBJahkKkqSWoSBJak1kKHjxmiT1YyJDwYvXJKkfExkKkqR+GAqSpJahIElqGQqSpJahIElqGQqSpJahIElqGQqSpJahIElqGQqSpNZEhoJzH0lSPyYyFJz7SJL6MZGhIEnqh6EgSWoZCpKklqEgSWoZCpKklqEgSWoZCpKklqEgSWoZCpKklqEgSWoZCpKklqEgSWoZCpKklqEgSWrNGgpJ3phkj+b5mUk+neQl/ZcmSRq1LnsK76uqB5K8Avhl4JPAef2WJUkahy6h8Gjz83XAeVX1GWDXPopJ8vokH0vymSSH97ENSdL0uoTC95J8FDgOuCzJEzsuB0CSTyS5J8mN27UfkeTWJLclOR2gqi6pqpOBE4E3df4tJEmLosuX+3HA5cARVfUvwNOA98xhG+uBI4YbkuwMfAg4EjgYOD7JwUMfObN5X5I0QrOGQlX9CLgHeEXT9Ajwna4bqKorgX/arvllwG1VdXtVPQxcCBydgXOAz1XVtVOtL8naJJuTbL733nu7liFJ6qDL2UdnAacBZzRNuwD/e4Hb3Qe4a+j1lqbtncCrgWOTnDLVglW1rqpWV9Xqvffee4FlSJKGPaHDZ94AvBi4FqCq7t52iuoCZIq2qqpzgXMXuG5J0jx1GVN4uKoKKIAkuy/CdrcA+w293he4exHWK0lagC6hcFFz9tFTk5wMfBH42AK3uwk4MMkBSXYF3gxc2nXhJGuSrNu6desCy5AkDZsxFJIE+GtgA/Ap4CDg/VX1p103kOQC4CrgoCRbkpxUVY8ApzI4q+kW4KKquqnrOqtqY1WtXbFiRddFJEkdzDimUFWV5JKqeinwhflsoKqOn6b9MuCy+axTktSPLoePvp7kkN4rmQMPH0lSP7qEwmHAVUn+Mck3k3wryTf7LmwmHj6SpH50OSX1yN6rkCQtCV1CoXqvQpK0JHQJhc8yCIYATwIOAG4F/n2Pdc0oyRpgzapVq8ZVgiQtS13mPnpBVb2w+Xkgg3mLvtp/aTPW5JiCJPVgzrfjbCaqW1JnI0mSFsesh4+S/M7Qy52AlwBOTypJy1CXMYXhye8eYTDG8Kl+ypEkjVOXULi5qi4ebkjyRuDiaT7fOweaJakfXcYUzujYNjIONEtSP6bdU0hyJPBaYJ8kw/c42JPBYSRJ0jIz0+Gju4HNwFHANUPtDwD/tc+iJEnjMW0oVNUNwA1Jzq+qfxthTZKkMekyprAyyYYkNye5fduj98pm4CypktSPLqHw58B5DMYRDgP+AvjLPouajQPNktSPLqGwW1V9CUhV3VlVHwB+qd+yJEnj0OU6hR8n2Qn4TpJTge8Bz+i3LEnSOHTZU3gX8GTgt4CXAm8Ffr3PoiRJ4zHrnkJVbQJIUlX1tv5LkiSNy6x7Ckl+IcnNwC3N659L8uHeK5MkjVyXw0d/DPwy8ENor194ZZ9FzcZTUiWpH53up1BVd23X9GgPtXTmKamS1I8uZx/dleTlQCXZlcGA8y39liVJGocuewqnAO8A9gG2AC9qXkuSlpmZZkk9p6pOAw6rqreMsCZJ0pjMtKfw2iS7MOZ7J0iSRmemMYW/A+4Ddk9yPxCgtv2sqj1HUJ8kaYSm3VOoqvdU1Qrgs1W1Z1XtMfxzhDVKkkZk1oHmqjp6FIVIksavyympS06SNcCaVatWjbsUadGdf/V3e13/CYfu3+v6Ndk6Xby21HjxmiT1Y9pQSPKl5uc5oytHkjROMx0+emaSXwSOSnIhg7OOWlV1ba+VSZJGbqZQeD9wOrAv8EfbvVd49zVJWnamDYWq2gBsSPK+qvrgCGuSJI1Jl5vsfDDJUTw+XfYVVfW3/ZYlSRqHLjfZORv4beDm5vHbTZskaZnpcp3C64AXVdVjAEk+CVyHcyJJ0rLT9TqFpw499+IASVqmuuwpnA1cl+TLDE5LfSXuJUjSstRloPmCJFcAhzAIhdOq6v/1XZikfixkGg2nyFj+Os19VFXfBy7tuZbOnPtIkvrh3EeSpNZEhoIkqR8zhkKSnZLcOKpiJEnjNWMoNNcm3JDE0SVJ2gF0GWh+JnBTkm8AD21rrKqjeqtKkjQWXULhd3uvQpK0JHS5TuErSZ4NHFhVX0zyZGDn/kuTJI1alwnxTgY2AB9tmvYBLumzKEnSeHQ5JfUdwH8E7geoqu8Az+izKEnSeHQJhZ9U1cPbXiR5AoM7r0mSlpkuofCVJP8d2C3Ja4CLgY39liVJGocuoXA6cC/wLeDtwGXAmX0WJUkajy5nHz3W3FjnagaHjW6tKg8fSdIyNGsoJHkd8BHgHxlMnX1AkrdX1ef6Lk6SNFpdLl77Q+CwqroNIMlzgc8ChoIkLTNdQuGebYHQuB24p6d6JC1hXW/Q4814Jte0oZDkmObpTUkuAy5iMKbwRmDTCGqTJI3YTHsKa4ae/wD4xeb5vcC/W+xCkjwHeC+woqqOXez1S5JmN20oVNXbFrryJJ8AfoXBIaj/MNR+BPAnDOZQ+nhV/c+quh04KcmGhW5XkjQ/Xc4+OgB4J7By+PMdp85eD/wv4C+G1rcz8CHgNcAWYFOSS6vq5rkULklafF0Gmi8B/ozBVcyPzWXlVXVlkpXbNb8MuK3ZMyDJhcDRQKdQSLIWWAuw//4OZknSYuoSCj+uqnMXcZv7AHcNvd4CHJrk6cDvAS9OckZVnT3VwlW1DlgHsHr1ai+ik6RF1CUU/iTJWcDngZ9sa6yqa+e5zUzRVlX1Q+CUea5TkrQIuoTCC4BfBX6Jxw8fVfN6PrYA+w293he4ey4rSLIGWLNq1ap5liBJmkqXUHgD8Jzh6bMXaBNwYDOA/T3gzcAJc1lBVW0ENq5evfrkRapJkkS3WVJvAJ46n5UnuQC4CjgoyZYkJ1XVI8CpwOXALcBFVXXTfNYvSVpcXfYUfgb4dpJN/PSYwqynpFbV8dO0X8ZgCm5J0hLSJRTO6r2KOXJMQVraZpsjybmRlq4u91P4yigKmQvHFCSpH12uaH6Ax+/JvCuwC/BQVe3ZZ2GSpNHrsqewx/DrJK9ncFWyJGmZ6XL20U+pqkuY/zUKiyLJmiTrtm7dOs4yJGnZ6XL46JihlzsBq3n8cNJYOKYgSf3ocvbR8H0VHgHuYDCBnSRpmekyprDg+ypIkibDTLfjfP8My1VVfbCHeiRJYzTTQPNDUzwATgJO67muGTnQLEn9mDYUquoPtz0Y3L9gN+BtwIXAc0ZU33S1bayqtStWrBhnGZK07Mw4ppDkacDvAG8BPgm8pKr+eRSFSZJGb6Yxhd8HjmGwl/CCqnpwZFVJksZipjGF/wY8CzgTuDvJ/c3jgST3j6Y8SdIoTbunUFVzvtp5VJwlVZL6sWS/+GfiQLMk9WMiQ0GS1A9DQZLUMhQkSS1DQZLUMhQkSa2JDAXnPpKkfkxkKHhKqiT1YyJDQZLUD0NBktQyFCRJLUNBktQyFCRJLUNBktQyFCRJrRlvx7lUeT8FabKdf/V3p2w/4dD9R1yJtjeRewpevCZJ/ZjIUJAk9cNQkCS1DAVJUstQkCS1DAVJUstQkCS1DAVJUstQkCS1DAVJUstQkCS1nPtI0pIx3ZxIs3HOpMUzkXsKzn0kSf2YyFCQJPXDUJAktQwFSVLLUJAktQwFSVLLUJAktQwFSVLLUJAktQwFSVLLUJAktQwFSVLLUJAktQwFSVLLUJAktQwFSVLLUJAktZbMndeS7A58GHgYuKKq/mrMJUnSDqfXPYUkn0hyT5Ibt2s/IsmtSW5LcnrTfAywoapOBo7qsy5J0tT6Pny0HjhiuCHJzsCHgCOBg4HjkxwM7Avc1Xzs0Z7rkiRNodfDR1V1ZZKV2zW/DLitqm4HSHIhcDSwhUEwXM8MYZVkLbAWYP/9vVm3JDj/6u9O+94Jh+4/6+eGPzPduqdaz2xtXerdfh0nHLp/+3Om5btsZz7GMdC8D4/vEcAgDPYBPg385yTnARunW7iq1lXV6qpavffee/dbqSTtYMYx0Jwp2qqqHgLeNupiJEmPG8eewhZgv6HX+wJ3z2UFSdYkWbd169ZFLUySdnTjCIVNwIFJDkiyK/Bm4NK5rKCqNlbV2hUrVvRSoCTtqPo+JfUC4CrgoCRbkpxUVY8ApwKXA7cAF1XVTX3WIUnqpu+zj46fpv0y4LI+ty1JmruJnObCMQVJ6sdEhoJjCpLUj4kMBUlSP1JV465h3pLcC9w5z8VXAPM9/tR12dk+N937U7V3adv+9V7AfR3qnC/7cOHsw4WbtD6cqn3Uffjsqpr66t+q2iEfwLq+l53tc9O9P1V7l7YpXm+2D+1D+3Bp9WHHPhtpHw4/duTDR9NOpbGIy872uenen6q9S9tCfqf5sA8Xzj5cuEnrw6nax92HrYk+fKSZJdlcVavHXccksw8Xzj5cuFH24Y68p7AjWDfuApYB+3Dh7MOFG1kfuqcgSWq5pyBJahkKkqSWoSBJahkKO5Akz0nyZ0k2jLuWSZXk9Uk+luQzSQ4fdz2TKMnzk3wkyYYkvznueiZVkt2TXJPkVxZzvYbChEvyiST3JLlxu/Yjktya5LYkpwNU1e1VddJ4Kl265tiHl1TVycCJwJvGUO6SNMc+vKWqTgGOAzxVtTGXPmycBly02HUYCpNvPXDEcEOSnYEPAUcCBwPHJzl49KVNjPXMvQ/PbN7XwHrm0IdJjgK+CnxptGUuaevp2IdJXg3cDPxgsYswFCZcVV0J/NN2zS8Dbmv2DB4GLgSOHnlxE2IufZiBc4DPVdW1o651qZrrv8OqurSqXg68ZbSVLl1z7MPDgJ8HTgBOTrJo3+W93mRHY7MPcNfQ6y3AoUmeDvwe8OIkZ1TV2WOpbjJM2YfAO4FXAyuSrKqqj4yjuAkx3b/DVwHHAE/Em23NZso+rKpTAZKcCNxXVY8t1gYNheUpU7RVVf0QOGXUxUyo6frwXODcURczoabrwyuAK0ZbysSasg/bJ1XrF3uDHj5anrYA+w293he4e0y1TCr7cOHsw4UbeR8aCsvTJuDAJAck2RV4M3DpmGuaNPbhwtmHCzfyPjQUJlySC4CrgIOSbElyUlU9ApwKXA7cAlxUVTeNs86lzD5cOPtw4ZZKHzohniSp5Z6CJKllKEiSWoaCJKllKEiSWoaCJKllKEiSWoaCxirJo0muH3qsHHdNS1GS5zX9c12S585j+XcleXIftWl58ToFjVWSB6vqKdO8Fwb/Rhdtsq9JkWTnqnp06PXpwG5VddY813cHsLqq7pvDMk9oLp7SDsQ9BS0pSVYmuSXJh4Frgf2SvCfJpiTfTPK7Q599b3PzkS8muSDJu5v2K5Ksbp7v1XwhkmTnJL8/tK63N+2vapbZkOTbSf6qCSSSHJLk/yS5Ick3kuyR5B+SvGiojq8leeF2v8eJGdyd7e+aGs8aeu+tzbquT/LRZs58kjyY5H8kuRr4haHPvxZ4F/Bfknx5lnWcl2Rzkpu29VWS3wKeBXx5aPkHh9Z/bJL1zfP1Sf6o+dw5SZ7b/A7XNL/38+b/X1cToap8+BjbA3gUuL55/A2wEngM+Pnm/cOBdQxmi9wJ+FvglcBLgW8BTwb2BG4D3t0scwWDv4oB9gLuaJ6vBc5snj8R2AwcALwK2MpgsrGdGEw18ApgV+B24JBmmT0ZzCz868AfN20/C2ye4vc6Efg+8HRgN+BGBncZez6wEdil+dyHgV9rnhdw3DT99IGh32+mdTyt+blz0w8vbF7fAew1tL4Hh54fC6xvnq9v+njn5vWXgAOb54cCfz/ufzM++n04dbbG7V+raviv7pXAnVX19abp8OZxXfP6KcCBwB7A31TVj5rlukwSdjjwwiTHNq9XNOt6GPhGVW1p1nU9g3DaCny/qjYBVNX9zfsXA+9L8h7gNxh8kU7lCzWYrpwkn2YQNI8wCLRNzc7IbsA9zecfBT7V4ff4TzOs47gkaxmE1zMZ3K3rmx3WOeziqno0yVOAlwMXN9uBQZhqGTMUtBQ9NPQ8wNlV9dHhDyR5F0Pzym/nER4/NPqk7db1zqq6fLt1vQr4yVDTowz+38hU26iqHyX5AoM7YM10n+Htl61mnZ+sqjOm+PyPa2gcYQZTriPJAcC7GezZ/HNzSOhJUyy/fW3bf2Zb/+8E/MtwaGv5c0xBS93lwG80f7WSZJ8kzwCuBN6QZLckewBrhpa5g8Ff0jA4NDK8rt9Mskuzrp9NsvsM2/428KwkhzSf3yPJtj+kPs7gZjubqmr7Wyhu85okT0uyG/B64GsMDscc2/wONO8/e/Zu+CnTrWNPBl/oW5P8DIP7+m7zAIO9q21+kOT5GdzG8Q1TbaTZM/q/Sd7YbCdJfm6OtWrCuKegJa2qPp/k+cBVzSGMB4G3VtW1Sf6awVjEncA/DC32B8BFSX4V+Puh9o8zOCx0bTOQfC+DL+vptv1wkjcBf9p8sf8rg1txPlhV1yS5H/jzGcr/KvCXwCrg/KraDJDkTODzzRfyvwHvaH6HTqrq5qnWUVVfT3IdcBODsZCvDS22Dvhcku9X1WHA6QzGDu5iMN4x5RlgDO6hfF6zvV0Y3CP4hq61avJ4SqqWhSQfYPBl/Qcj2t6zGAzkPq+mOGU2g3vnrq7mXrrSpPDwkTRHSX4NuBp471SBIE0y9xQkSS33FCRJLUNBktQyFCRJLUNBktQyFCRJLUNBktT6/6N3a+049wa8AAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "table   = Artifact.load(workdir+'/feature_table/table.qza')\n",
    "rep_seq = Artifact.load(workdir+'/rep_seq/rep_seq.qza')\n",
    "output_viz = feature_table.visualizers.summarize(table,sample_metadata)\n",
    "output_viz.visualization.save(workdir+'/feature_table/table.qzv')\n",
    "output_viz.visualization"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate a tree for phylogenetic diversity analyses"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: mafft --preservecase --inputorder --thread 1 /var/folders/bj/4d5l6xgx2fjgpp5jy9ql6jfh0000gn/T/qiime2-archive-7z2wq84t/a1d69b1f-7894-463e-8c76-6d2b65724420/data/dna-sequences.fasta\n",
      "\n",
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: FastTree -quote -nt /var/folders/bj/4d5l6xgx2fjgpp5jy9ql6jfh0000gn/T/qiime2-archive-vz7he0jn/872733ae-e70e-4784-bdc0-94ffc9f660d8/data/aligned-dna-sequences.fasta\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'/Users/alchemist/Aakash/BonnGut/Microbiome_pipeline/phylo_tree/rooted_tree.qza'"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phylogenic_tree = phylogeny.pipelines.align_to_tree_mafft_fasttree(rep_seq)#,n_threads='auto')\n",
    "phylogenic_tree.rooted_tree.save(workdir+'/phylo_tree/rooted_tree.qza')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Taxonomy Assignment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "/Users/alchemist/Aakash/BonnGut/Microbiome_pipeline/taxonomy/classifier/classifier_ezbiocloud.qza does not exist.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-21-e4fda4a7e09c>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mgg_classifier\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mArtifact\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mload\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mworkdir\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m'/taxonomy/classifier/classifier_ezbiocloud.qza'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/qiime2/sdk/result.py\u001b[0m in \u001b[0;36mload\u001b[0;34m(cls, filepath)\u001b[0m\n\u001b[1;32m     64\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mload\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcls\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfilepath\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     65\u001b[0m         \u001b[0;34m\"\"\"Factory for loading Artifacts and Visualizations.\"\"\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 66\u001b[0;31m         \u001b[0marchiver\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0marchive\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mArchiver\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mload\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     67\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     68\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mArtifact\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_is_valid_type\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marchiver\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/qiime2/core/archive/archiver.py\u001b[0m in \u001b[0;36mload\u001b[0;34m(cls, filepath)\u001b[0m\n\u001b[1;32m    297\u001b[0m     \u001b[0;34m@\u001b[0m\u001b[0mclassmethod\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    298\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mload\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcls\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfilepath\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 299\u001b[0;31m         \u001b[0marchive\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcls\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_archive\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    300\u001b[0m         \u001b[0mFormat\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcls\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_format_class\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marchive\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mversion\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    301\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mFormat\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/qiime2-2020.8/lib/python3.6/site-packages/qiime2/core/archive/archiver.py\u001b[0m in \u001b[0;36mget_archive\u001b[0;34m(cls, filepath)\u001b[0m\n\u001b[1;32m    257\u001b[0m         \u001b[0mfilepath\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpathlib\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mPath\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    258\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mfilepath\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexists\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 259\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"%s does not exist.\"\u001b[0m \u001b[0;34m%\u001b[0m \u001b[0mfilepath\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    260\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    261\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0m_ZipArchive\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mis_archive_type\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: /Users/alchemist/Aakash/BonnGut/Microbiome_pipeline/taxonomy/classifier/classifier_ezbiocloud.qza does not exist."
     ]
    }
   ],
   "source": [
    "gg_classifier = Artifact.load(workdir+'/taxonomy/classifier/classifier_ezbiocloud.qza')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/alchemist/Aakash/BonnGut/Hannahs_study/taxonomy/taxonomy_ezbiocloud.qza'"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "taxonomy = feature_classifier.methods.classify_sklearn(reads = rep_seq,\n",
    "                                                       classifier = gg_classifier)\n",
    "taxonomy.classification.save(workdir+'/taxonomy/taxonomy_ezbiocloud.qza')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/var/folders/bj/4d5l6xgx2fjgpp5jy9ql6jfh0000gn/T/qiime2-archive-ok8pv6cl')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 7f9a1737-eb7d-4b88-8599-fa34df549c37>"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "taxonomy_classification = Artifact.load(workdir+'/taxonomy/taxonomy_ezbiocloud.qza')\n",
    "taxonomy_vis = metadata.visualizers.tabulate(taxonomy_classification.view(qiime2.Metadata))\n",
    "taxonomy_vis.visualization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "taxa_bar_plot = taxa.visualizers.barplot(table, taxonomy_classification, sample_metadata)\n",
    "taxa_bar_plot.visualization.export_data(workdir+'/taxonomy')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
